### PR TITLE
Fix #<1444> - <Sync Survey dialog box Bug>

### LINKED ISSUE #1444: Sync Survey dialog box Bug
**Summary:** 

On hiding Sync Survey dialog box, it should disappear from the main screen and should appear in the notification panel. However, it disappears entirely.

**Steps to reproduce:** 
1. Goto Settings in Online Mode
2. Toggle Sync Surveys on
3. Hide the Dialog box which appears by clicking the hide button

**Expected behaviour:** 

On hiding the dialog box, it should disappear from main activity and should appear in notification panel.

**Observed behaviour:** 

The dialog box disappears completely on clicking the hide button.

**Device and Android version:** 

Device - Samsung Galaxy S10e
Android Version - 10 (Android Q)
Custom ROM - Samsung One UI version 2.0

**Screenshots:** 

screenshot

### PR DESCRIPTION
**Pull request by Sachit1137**
Fixes #1444 

Please Add Screenshots If there are any UI changes.
ezgif com-video-to-gif

Please make sure these boxes are checked before submitting your pull request - thanks!

- [x] Apply the `MifosStyle.xml` style template to your code in Android Studio.

- [x] Run the unit tests with `./gradlew check` to make sure you didn't break anything

- [x] If you have multiple commits please combine them into one commit by squashing them.

### CODE REVIEW

File: mifosng-android/src/main/java/com/mifos/mifosxdroid/dialogfragments/syncsurveysdialog/SyncSurveysDialogFragment.java [hunk 1]
```diff
@@ -32,6 +40,7 @@
 public class SyncSurveysDialogFragment extends DialogFragment implements SyncSurveysDialogMvpView {
 
     public static final String LOG_TAG = SyncSurveysDialogFragment.class.getSimpleName();
+    public static final String CHANNEL_ID = "1001";
```
Comment (Grandolf49) [inline mifosng-android/src/main/java/com/mifos/mifosxdroid/dialogfragments/syncsurveysdialog/SyncSurveysDialogFragment.java]: Avoid using Hard Coded strings in the code. Create a value in strings.xml and use it from there

File: mifosng-android/src/main/res/values/strings.xml [hunk 2]
```diff
@@ -199,6 +199,8 @@
     <string name="add_payment_detail">Add Payment Detail</string>
     <string name="generate_new">Generate New</string>
     <string name="currency">Currency</string>
+    <string name="Channel_Name">Channel_Name</string>
+    <string name="Channel_Description">Channel_Description</string>
```
Comment (Grandolf49) [inline mifosng-android/src/main/res/values/strings.xml]: Where are these 2 strings being used in your code?

Comment (Sachit1137) [reply mifosng-android/src/main/res/values/strings.xml]: These are being used in the method- createNotificationChannel() in class SyncSurveyDialogFragment Line number 106 and 107.

File: mifosng-android/src/main/java/com/mifos/mifosxdroid/dialogfragments/syncsurveysdialog/SyncSurveysDialogFragment.java (hunk 1 above)
Comment (Sachit1137) [reply mifosng-android/src/main/java/com/mifos/mifosxdroid/dialogfragments/syncsurveysdialog/SyncSurveysDialogFragment.java]: Ok. Will take care of it. Thanks for telling me.

Comment (Sachit1137) [reply mifosng-android/src/main/java/com/mifos/mifosxdroid/dialogfragments/syncsurveysdialog/SyncSurveysDialogFragment.java]: @Grandolf49 I have made the required changes.

**Comment by Sachit1137** [comment]
@RajaVamsi11 @Grandolf49  I tried squashing my commits. But I ran into an error and now I am not able to publish my branch to Github.

S1

As you can see above my three commits have been squashed into one.

S2

But when I publish my branch to Github. The above error comes. I tried fetching but the error keeps on coming again and again.

Kindly help me on this.

**Comment by RajaVamsi11** [comment]
@Sachit1137 Did you try force push? In my case I always work with GIT to make PRs and edit them.

**Comment by RajaVamsi11** [comment]
@Grandolf49 Do you any idea regarding above issue ?

**Comment by Grandolf49** [comment]
@RajaVamsi11 since the remote is not in sync with the local, git fetch needs to be performed before pushing the commits

**Comment by Sachit1137** [comment]
@RajaVamsi11 I want to force push. The command for the same is **git push --force origin master** ?

**Comment by Sachit1137** [comment]
@RajaVamsi11  @Grandolf49 So what is happening is when I force push after squashing the commits, besides my commit some previous commits also get force pushed. How can I solve this?

**Comment by Grandolf49** [comment]
@Sachit1137 I had an issue similar like this once. I resolved it by rebasing my branch to the master branch and commiting the changes again. I know it's not an optimized solution but worked for me.

**Comment by Sachit1137** [comment]
@Grandolf49 Will try doing that. Thanks

**Comment by AbhilashG97** [comment]
Hello @Sachit1137, 

Please squash your commits into a single commit. I'm closing this pull request for now.

Please re-open this pull request after you have made the suggested changes.

Thank you for your contribution.

**Pull request closed without merging**